### ISSUE TITLE
Causes ClassNotFoundException in Sponge 8

### ISSUE THREAD

**Issue by pschichtel** [lucko/spark#124]
Not sure since when this is happening, but we deployed a newer snapshots today and this started happening during server startup:

```
2021-07-03T16:08:45,849 INFO [main] [launch]: Loading Minecraft Server, please wait...
2021-07-03T16:08:46,529 INFO [main] [c.moj.aut.ygg.YggdrasilAuthenticationService]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[16:08:47] [main/ERROR]: Failed to invoke main class org.spongepowered.vanilla.applaunch.Main due to an error: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:39)
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54)
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72)
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82)
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66)
	at org.spongepowered.vanilla.applaunch.Main.run(Main.java:79)
	at org.spongepowered.vanilla.applaunch.Main.main(Main.java:56)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.spongepowered.vanilla.installer.InstallerMain.invokeMain(InstallerMain.java:150)
	at org.spongepowered.vanilla.installer.InstallerMain.downloadAndRun(InstallerMain.java:133)
	at org.spongepowered.vanilla.installer.InstallerMain.run(InstallerMain.java:75)
	at org.spongepowered.vanilla.installer.InstallerMain.main(InstallerMain.java:70)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.spongepowered.vanilla.applaunch.handler.prod.ServerProdLaunchHandler.launchService0(ServerProdLaunchHandler.java:45)
	at org.spongepowered.vanilla.applaunch.handler.AbstractVanillaLaunchHandler.lambda$launchService$1(AbstractVanillaLaunchHandler.java:172)
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37)
	... 14 more
Caused by: java.lang.NoClassDefFoundError: org/jetbrains/annotations/ApiStatus
	at java.lang.Class.getDeclaringClass0(Native Method)
	at java.lang.Class.getDeclaringClass(Class.java:1235)
	at java.lang.Class.getEnclosingClass(Class.java:1277)
	at sun.reflect.annotation.AnnotatedTypeFactory.addNesting(AnnotatedTypeFactory.java:96)
	at sun.reflect.annotation.AnnotatedTypeFactory.buildAnnotatedType(AnnotatedTypeFactory.java:65)
	at sun.reflect.annotation.TypeAnnotationParser.buildAnnotatedTypes(TypeAnnotationParser.java:137)
	at sun.reflect.annotation.TypeAnnotationParser.buildAnnotatedInterfaces(TypeAnnotationParser.java:186)
	at java.lang.Class.getAnnotatedInterfaces(Class.java:3636)
	at io.leangen.geantyref.GenericTypeReflector.getExactDirectSuperTypes(GenericTypeReflector.java:529)
	at io.leangen.geantyref.GenericTypeReflector.getExactSuperType(GenericTypeReflector.java:261)
	at io.leangen.geantyref.GenericTypeReflector.isSuperType(GenericTypeReflector.java:394)
	at io.leangen.geantyref.TypeFactory.checkParametersWithinBound(TypeFactory.java:209)
	at io.leangen.geantyref.TypeFactory.parameterizedInnerClass(TypeFactory.java:162)
	at io.leangen.geantyref.TypeFactory.parameterizedClass(TypeFactory.java:45)
	at org.spongepowered.common.data.key.SpongeKeyBuilder.elementType(SpongeKeyBuilder.java:79)
	at org.spongepowered.api.data.Keys.key(Keys.java:2731)
	at org.spongepowered.api.data.Keys.<clinit>(Keys.java:381)
	at org.spongepowered.common.data.provider.block.location.ServerLocationData.register(ServerLocationData.java:50)
	at org.spongepowered.common.data.provider.block.location.LocationDataProviders.registerProviders(LocationDataProviders.java:33)
	at org.spongepowered.common.data.provider.DataProviderRegistratorBuilder.register(DataProviderRegistratorBuilder.java:36)
	at org.spongepowered.common.data.provider.DataProviderRegistry.registerDefaultProviders(DataProviderRegistry.java:225)
	at org.spongepowered.common.data.provider.DataProviderRegistry.registerDefaultProviders(DataProviderRegistry.java:209)
	at org.spongepowered.common.data.SpongeDataManager.registerDefaultProviders(SpongeDataManager.java:318)
	at org.spongepowered.common.SpongeLifecycle.establishDataProviders(SpongeLifecycle.java:113)
	at net.minecraft.server.Main.redirect$zco000$impl$configurePackRepository(SourceFile:577)
	at net.minecraft.server.Main.main(SourceFile:131)
	at org.spongepowered.vanilla.launch.DedicatedServerLaunch.lambda$performBootstrap$0(DedicatedServerLaunch.java:53)
	at org.spongepowered.common.SpongeBootstrap.perform(SpongeBootstrap.java:61)
	at org.spongepowered.vanilla.launch.DedicatedServerLaunch.performBootstrap(DedicatedServerLaunch.java:53)
	at org.spongepowered.vanilla.launch.VanillaLaunch.launchPlatform(VanillaLaunch.java:111)
	at org.spongepowered.vanilla.launch.DedicatedServerLaunch.launch(DedicatedServerLaunch.java:43)
	... 21 more
Caused by: java.lang.ClassNotFoundException: org.jetbrains.annotations.ApiStatus
	at java.lang.ClassLoader.findClass(ClassLoader.java:523)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:106)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 52 more
```

This only happens with spark. I even tried building a fresh build from latest master, but no difference.

**Comment by pschichtel** [comment]
This is on adoptopenjdk 8

**Comment by lucko** [comment]
Fixed in 3c12a31ead109c11af7290d13e87651cff796b6c
